### PR TITLE
[Merged by Bors] - TO-3326 don't return interacted documents

### PR DESCRIPTION
**Pull request by janpetschexain**
**References**

- [TO-3326]

**Summary**

- fix index naming
- add getter for interacted doc ids in postgres
- add filter for excluded doc ids in elastic

[TO-3326]: https://xainag.atlassian.net/browse/TO-3326?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine_core/web-api/src/migrations/20220926145400_document.sql [hunk 1]
```diff
@@ -20,5 +20,5 @@ CREATE TABLE IF NOT EXISTS interaction (
     PRIMARY KEY (doc_id, user_id, time_stamp)
 );
 
-CREATE INDEX IF NOT EXISTS idx_doc_by_user_id
-    ON document(user_id);
+CREATE INDEX IF NOT EXISTS idx_interaction_by_user_id
+    ON interaction(user_id);
```
Comment (biernacki) [inline discovery_engine_core/web-api/src/migrations/20220926145400_document.sql:24]: yeah, I also noticed that after rebasing :)

File: discovery_engine_core/web-api/src/storage.rs [hunk 2]
```diff
@@ -171,6 +172,27 @@ impl UserState {
...
+
+    #[allow(dead_code)]
+    pub(crate) async fn interacted_documents(
+        &self,
+        user_id: &UserId,
+    ) -> Result<Vec<DocumentId>, GenericError> {
+        let mut tx = self.pool.begin().await?;
+
+        let documents = sqlx::query_as::<_, QueriedInteractedDocumentId>(
+            "SELECT doc_id
```
Comment (biernacki) [inline discovery_engine_core/web-api/src/storage.rs:184]: if I understand correctly `SELECT DISTINCT` will get rid off duplicates and our PK is (doc_id, user_id, time_stamp) which means in theory we could have same doc_id multiple times returned, yes?

**Review by biernacki — COMMENTED** [review]
I reviewed it only because I was curios if there is already a fix for the index in interaction table.

File: discovery_engine_core/web-api/src/storage.rs (hunk 2 above)
Comment (janpetschexain) [reply discovery_engine_core/web-api/src/storage.rs:184]: good point, it shouldn't matter much if we feed duplicates to the filter in elastic, but it's definitely better to return unique values here, i'll change it.

File: discovery_engine_core/web-api/src/migrations/20220926145400_document.sql (hunk 1 above)
Comment (janpetschexain) [reply discovery_engine_core/web-api/src/migrations/20220926145400_document.sql:24]: yeah, sorry for that, i must have messed this up during one of my rebases.

**Review by biernacki — APPROVED** [review]
lgtm

**Comment by janpetschexain** [comment]
bors merge

**Pull request closed without merging**